modules/features2d/src/orb.cpp

```diff
@@ -1036,7 +1036,11 @@ void ORB_Impl::detectAndCompute( InputArray _image, InputArray _mask,
     bool useOCL = false;
 #endif
 
-    Mat image = _image.getMat(), mask = _mask.getMat();
+    Mat image = _image.getMat(), mask;
+    if (!_mask.empty())
+    {
+        cv::threshold(_mask.getMat(), mask, 0, 255, cv::THRESH_BINARY);
```

Comment:
threshold(_mask, mask, 0, 255, THRESH_BINARY);

Reply:
Your snippet does the same thing but in a concise way:), do you want me to implement this change?

Reply:
yes exactly does the same thing.
no need to `cv::` prefix and `.getMat()`